CMakeLists.txt

```diff
@@ -250,9 +262,19 @@ if(HARDENING)
       $<$<NOT:$<CONFIG:Debug>>:-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3>
     )
 
-    try_append_cxx_flags("-Wstack-protector" TARGET hardening)
-    try_append_cxx_flags("-fstack-protector-all" TARGET hardening)
     try_append_cxx_flags("-fcf-protection=full" TARGET hardening)
+    try_append_cxx_flags("-fstack-protector-all" TARGET hardening)
+    try_append_cxx_flags("-Wstack-protector" TARGET hardening)
+
+    try_append_linker_flag("-fstack-protector-all" TARGET hardening)
+
+    try_append_linker_flag("-Wl,--enable-reloc-section" TARGET hardening)
+    try_append_linker_flag("-Wl,--dynamicbase" TARGET hardening)
+    try_append_linker_flag("-Wl,--nxcompat" TARGET hardening)
+    try_append_linker_flag("-Wl,--high-entropy-va" TARGET hardening)
+    try_append_linker_flag("-Wl,-z,relro" TARGET hardening)
+    try_append_linker_flag("-Wl,-z,now" TARGET hardening)
+    try_append_linker_flag("-Wl,-z,separate-code" TARGET hardening)
 
     if(MINGW)
```

Comment:
Not exactly related to this change, but when reading over this I keep thinking it would be good if we could have some sort of "not Windows" condition, and avoid conditionals like this with empty happy-path branches. It seems in general, Windows is the "first class citizen" in this CMake setup, with all conditions being "if windows/MSVC else everything else". It does seem a little bit backwards.

Reply:
Something like the [`WIN32`](https://cmake.org/cmake/help/latest/variable/WIN32.html) variable?